lib/hooks/userspace/hook-koa.js

```diff
@@ -0,0 +1,130 @@
+/**
```

Comment:
Meta: since this file is using generators already, should it use more ES6 features?